staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,408 @@
+			}
+		}
+	}
+
+	return nil
+}
+
+// applyAliases firstly appends all defined aliases in kuberc file to the root command.
+// Since there may be several alias definitions belonging to the same command, it extracts the
+// alias that is currently executed from args. After that it sets the flag definitions in alias as default values
+// of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
+// is appended into the command args.
+func (p *Preferences) applyAliases(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) ([]string, error) {
+	_, _, err := rootCmd.Find(args[1:])
+	if err == nil {
+		// Command is found, no need to continue for aliasing
+		return args, nil
+	}
+
+	var aliasArgs *aliasing
+
+	var commandName string // first "non-flag" arguments
+	for _, arg := range args[1:] {
+		if !strings.HasPrefix(arg, "-") {
+			commandName = arg
+			break
+		}
+	}
+	commandName = strings.TrimSpace(commandName)
+
+	for _, alias := range kuberc.Aliases {
+		p.aliases[alias.Name] = struct{}{}
+		if alias.Name != commandName {
+			continue
+		}
+
+		// do not allow shadowing built-ins
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			fmt.Fprintf(errOut, "Warning: Setting alias %q to a built-in command is not supported\n", alias.Name)
+			break
+		}
+
+		commands := strings.Fields(alias.Command)
+		existingCmd, flags, err := rootCmd.Find(commands)
+		if err != nil {
+			return args, fmt.Errorf("command %q not found to set alias %q: %v", alias.Command, alias.Name, flags)
+		}
+
+		newCmd := *existingCmd
+		newCmd.Use = alias.Name
+		aliasCmd := &newCmd
+
+		aliasArgs = &aliasing{
+			args:    alias.Args,
+			flags:   alias.Flags,
+			command: aliasCmd,
+		}
+		break
+	}
+
+	if aliasArgs == nil {
+		// pursue with the current behavior.
+		// This might be a built-in command, external plugin, etc.
+		return args, nil
+	}
+
+	rootCmd.AddCommand(aliasArgs.command)
+
+	foundAliasCmd, _, err := rootCmd.Find([]string{commandName})
+	if err != nil {
+		return args, nil
+	}
+
+	// This function triggers merging the persistent flags in the parent commands.
+	_ = foundAliasCmd.InheritedFlags()
+
+	for _, fl := range aliasArgs.flags {
+		existingFlag := foundAliasCmd.Flag(fl.Name)
+		if existingFlag == nil {
+			return args, fmt.Errorf("invalid alias flag %s in alias %s", fl.Name, args[0])
+		}
+		if searchInArgs(existingFlag.Name, existingFlag.Shorthand, foundAliasCmd.Flags(), args) {
+			// Don't modify the value implicitly, if it is passed in args explicitly
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
```

Comment:
I think we may need prepend / append arg capability

Reply:
Sounds like an useful functionality and enrich the behavior and we can extend it easily exposing a new field `appendargs` as you recommend ^^. Wouldn't it be better to defer this to next releases and reconsider after getting some feedback from users?. I believe in any case we'll need to have v1beta1 release after seeing different use cases and we can combine this also with them?. WDYT?.

Reply:
knowing that there are use cases that require both types, I think we should go ahead and put both in the config and name the config fields clearly to avoid confusion from the start

Reply:
I've added to two new fields (in lieu of `args`) - `prependArgs` and `appendArgs`.
`prependArgs` will be inserted just after the alias name and `appendArgs` will be appended to the args directly. In order to ensure that functionality works, added multiple tests cases.

Reply:
Merge conflict is resolved and I believe all the suggestions are applied. @liggitt PTAL when you have a moment. Thank you.